### ISSUE TITLE
[feature] Support nested execution in `platform: system` commands

### ISSUE THREAD

**Issue by shellscape** [moonrepo/moon#889]
**Is your feature request related to a problem? Please describe.**

While trying to get around https://github.com/moonrepo/moon/issues/888 without creating a separate bash script, I attempted to use the `MOON_PROJECT_ROOT` environment variable in a command with `platform: system`.

```yml
  test:
    command: node --title $projectAlias:test ../../node_modules/ava/entrypoints/cli.mjs $(echo $MOON_PROJECT_ROOT)
    inputs:
      - src
      - test
    platform: system
```

The result was:

```
[DEBUG 11:35:05] moon_process::command_inspector  Running command /bin/zsh -c 'node --title svc-test:test ../../node_modules/ava/entrypoints/cli.mjs '\''$(echo'\'' '\''$MOON_PROJECT_ROOT)'\''' 
```

I also tried backticks with a similar result:

```yml
  test:
    command: node --title $projectAlias:test ../../node_modules/ava/entrypoints/cli.mjs `echo $MOON_PROJECT_ROOT`
    inputs:
      - src
      - test
    platform: system
```

It looks like the argument parsing needs a bit more massaging to make it bash-compatible yet, but that's to be expected.

**Describe the solution you'd like**

I'd like to use backticks or the `$()` blocks for nested execution. This likely requires closer integration with the shell and probably not something that moon is setup for at the moment, but I'd still like to get it down as a want and nice-to-have.

**Describe alternatives you've considered**

Using a separate bash script works, but that's a detriment to DX.

**Additional context**

**Comment by shellscape** [comment]
Ah that's right. That form totally skipped my mind.

**Comment by shellscape** [comment]
Good call. Gave that a shot and looks like the parser is making it a string, which doesn't allow the nested execution. `'\''$(echo $MOON_PROJECT_ROOT)'\'''` - it's no worries, I get you're limited at the moment. Good for tracking though.

**Comment by ilyasotkov** [comment]
Should this be considered a bug? I saw this example in the [documentation](https://moonrepo.dev/docs/config/project#shell):

```yaml
tasks:
  native:
    command: 'echo $SHELL'
    options:
      shell: true
```

which I think implies that it will print the name of the shell used and not just the string `$SHELL`.

Can work around with:

```yaml
tasks:
  native:
    command: bash -c "echo $SHELL"
    options:
      shell: true
```

**Comment by milesj** [comment]
This PR (https://github.com/moonrepo/moon/pull/928) will fix the quoting, so `$(echo $MOON_PROJECT_ROOT)` should work once it lands.

**Comment by milesj** [comment]
This should now be possible with task scripts https://moonrepo.dev/blog/moon-v1.27#new-task-scripts
